File: app/src/main/java/io/github/sithengineer/motoqueiro/network/NetworkService.java
package io.github.sithengineer.motoqueiro.network;

import io.github.sithengineer.motoqueiro.data.model.RidePart;
import retrofit2.http.Body;
import retrofit2.http.POST;
import rx.Observable;

public interface NetworkService {

  //
  // ride
  //
  @POST("ride") Observable<Void> upload(@Body RidePart ride);
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/network/NetworkService.kt
package io.github.sithengineer.motoqueiro.network

import io.github.sithengineer.motoqueiro.data.local.entity.Ride
import io.reactivex.Observable
import retrofit2.http.Body
import retrofit2.http.POST

interface NetworkService {

  @POST("ride")
  fun upload(@Body ride: Ride): Observable<Void>
}
